### ISSUE TITLE
Docs: Improve link documentation

### ISSUE THREAD

**Issue by Dioprz** [typst/typst#3846]
Hi,

Recently I was trying to create a show rule to display links with a custom color, and I was very confused about how to do it using an anonymous function. What I want to make is something like this (from the docs):

```typst
#show heading: it => [
  #set align(center)
  #set text(font: "Inria Serif")
  \~ #emph(it.body)
     #counter(heading).display(
       it.numbering
     ) \~
]
```

This doesn't work because I can't find the way to properly declare the link, without introduce a recursion:

```typst
#show links: it => link(it.dest)[#text(fill:blue, it.body)]
```
But if I remove the `link(it.dest)` part, the links ended up being just text.

I resigned myself and just did

```typst
#show links: text(fill:blue)
```

Now, however, seeing this https://github.com/typst/typst/discussions/888#discussioncomment-5670498 I learn that my error was to try to return just the `it.body`, instead of the `it`. So I want to ask, is this something we could document on the link function documentation, or the Styling/show rules documentation? Just add this example could help anyone else to understand that, in the case of the link function, you shouldn't detach both `it.dest` and `it.body` in the return.

I could do the changes if needed, just want to know where and if they will be welcome.

**Comment by laurmaedje** [comment]
Better documentation on using `it` directly somewhere in the [show rule section of the styling chapter](https://github.com/typst/typst/blob/main/docs/reference/styling.md#show-rules) sounds like a good idea. The styling chapter needs a revamp in general ...

**Comment by Dioprz** [comment]
Great, thanks for the quick response!

I will try to make some time this week to add the comment and see if I can improve anything else on the Styling chapter. 

However, I am lacking knowledge of the internals, so I will think of a few questions (while reading the chapter deeply) to ask you and clarify some things (if you don't mind, of course). Then I could try to rewrite the section. 

Sounds good to you?

**Comment by laurmaedje** [comment]
Sounds good.

**Comment by Dioprz** [comment]
Hello! I'm back to continue working on this issue. Sorry for the delay.

I've reviewed the Styling section and have some questions:

1. In several places, the word "element" is used but sometimes I'm not sure if it is used as a abbreviation of a *function element* or as a term to describe just *something*. An example of this are both first sentences of the *set rules* and *show rules* subsections, respectively:

    - With set rules, you can customize the appearance of elements. 

    -  With show rules, you can deeply customize the look of a type of element.

2. Terminology: "optional" vs. "settable" parameters: The Styling section uses "optional" to describe parameters within a set rule, but the linked documentation uses "settable." Could you confirm the preferred terminology?

3. I have a better grasp of how "it" is used, but I'd like more clarity on its capabilities and limitations depending on the function element or selector. Could you provide either some insights into how "it" works internally or point me to relevant code sections for further investigation?

I look forward to your answer to create a draft PR and start working on based on your feedback. Also, if there are specific areas within the Styling section you believe would benefit from a revamp, please let me know what you have in mind and I will try to work with that!

**Comment by laurmaedje** [comment]
> 1. In several places, the word "element" is used but sometimes I'm not sure if it is used as a abbreviation of a function element or as a term to describe just something. An example of this are both first sentences of the set rules and show rules subsections, respectively:

I'm not sure whether "element" is possibly sometimes used as a term for just something, but in principle it does mean a function element and it definitely does in the two sentences you quotes.

> 2. Terminology: "optional" vs. "settable" parameters: The Styling section uses "optional" to describe parameters within a set rule, but the linked documentation uses "settable." Could you confirm the preferred terminology?

I think it depends a bit on what we want to emphasize. Overall, I'd say that parameters can be optional and that _makes them_ settable. In that sense, settable is just a way to say that optional parameters can be used with set rules. But the more generic term is definitely "optional".

> 3. I have a better grasp of how "it" is used, but I'd like more clarity on its capabilities and limitations depending on the function element or selector. Could you provide either some insights into how "it" works internally or point me to relevant code sections for further investigation?

`it` generally is the element currently being shown and has two purposes:
- it provides the fields of the element
- if directly used as content in a show rule, it displays as the element would have without the current show rule (i.e. it takes into account the remaining, existing show rules & the default one).

Internally, `it` is just the `Content` that the compiler saw and executed the show rule for, but with an extra bit set to know that this show rule needn't be executed again for the element (so that if it is visited again when processing the content resulting from the show rule, the next outer show rule is triggered instead). 

> I look forward to your answer to create a draft PR and start working based on your feedback. Also, if there are specific areas within the Styling section you believe would benefit from a revamp, please let me know what you have in mind and I will try to work with that!

I think what we should definitely focus on is to better explain `show-set` rules (and how it compares to `show`) and how to use selectors (in particular `where`, but also others) with more examples. If you have any questions about these things, please let me know! cc: @PgBiel who is also knowledgeable about show-set.

**Comment by PgBiel** [comment]
I agree `show-set` needs more documentation, especially since its semantics changed fairly recently. We should emphasize that

```typ
#show x: it => {
  set y(a: b, c: d)
  it
}
```

cannot be overridden, as writing it again but with different values for the same parameters in `set y` would have no effect - that is, writing the same thing but with `set y(a: bbb, c: ddd)` wouldn't cause any visible changes -, whereas

```typ
#show x: set y(a: b, c: d)
```

can be overridden by a `#show x: set y(a: bbb, c: ddd)` later.
